use-cases/using-Intel-QAT-and-kata.md

```diff
@@ -0,0 +1,484 @@
+$ export KERNEL_SUBLEVEL=$(awk '/^SUBLEVEL =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_EXTRAVERSION=$(awk '/^EXTRAVERSION =/{print $NF}' $GOPATH/$LINUX_VER/Makefile)
+$ export KERNEL_ROOTFS_DIR=${KERNEL_MAJOR_VERSION}.${KERNEL_PATHLEVEL}.${KERNEL_SUBLEVEL}${KERNEL_EXTRAVERSION}
+
+$ cd $QAT_SRC
+$ sudo -E make clean
+
+$ KERNEL_SOURCE_ROOT=$GOPATH/$LINUX_VER ./configure --disable-qat-lkcf --enable-icp-sriov=guest
+$ sudo -E make quickassist-all -j$(nproc)
+$ sudo -E make INSTALL_MOD_PATH=$ROOTFS_DIR qat-driver-install -j$(nproc)
+
+# usdm_drv also needs to be copied into rootfs modules path and depmod run
+$ sudo cp $QAT_SRC/build/usdm_drv.ko $ROOTFS_DIR/usr/lib/modules/${KERNEL_ROOTFS_DIR}/updates/drivers  
+
+# Note: usdm_drv will be loaded by configuring kata kernel boot parameter in kata config
+# Ignore any errors on modules.builtin and modules.order
+$ sudo depmod -a -b ${ROOTFS_DIR} ${KERNEL_ROOTFS_DIR}
+
+# Create Kata rootfs
+$ cd ${OSBUILDER}/osbuilder/image-builder
+$ script -fec 'sudo -E USE_DOCKER=true ./image_builder.sh ${ROOTFS_DIR}'
+```
+
+### Copy Kata rootfs
+
+```sh
+$ mkdir -p $KATA_ROOTFS_LOCATION
+$ cp ${OSBUILDER}/osbuilder/image-builder/kata-containers.img $KATA_ROOTFS_LOCATION
+```
+
+### Update Kata configuration to point to custom kernel and rootfs
+
+You must update the `configuration.toml` for Kata Containers to point to the 
+custom kernel, custom rootfs, and to specify which modules to load when the 
+virtual machine is booted when a container is run. The example below assumes 
+an Intel QAT is installed and those modules need to be loaded.
+
+```sh
+$ sudo mkdir -p /etc/kata-containers
+$ sudo cp /usr/share/defaults/kata-containers/configuration-qemu.toml /etc/kata-containers/configuration.toml
+
+# Make sure usdm_drv is loaded in the image
+$ sudo sed -i "s|kernel_params = \"\"|kernel_params = \"modules-load=usdm_drv,qat_c62xvf\"|g" /etc/kata-containers/configuration.toml
+
+# Change the default kernel and rootfs
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/kata-containers.img|${KATA_KERNEL_LOCATION}\/kata-containers.img|g" /etc/kata-containers/configuration.toml
+$ sudo sed -i "s|\/usr\/share\/kata-containers\/vmlinuz.container|${KATA_ROOTFS_LOCATION}\/vmlinuz-${LINUX_VER}_qat|g" /etc/kata-containers/configuration.toml
+```
+
+## Verify Intel QAT works in a Docker Kata Containers container
+
+The following instructions leverage an OpenSSL Dockerfile that builds the 
+Intel QAT engine to allow OpenSSL to offload crypto functions. It is a 
+convenient way to test that VFIO device passthrough for the Intel QAT VF’s are
+working properly with the Kata Containers VM.
+
+## Build OpenSSL Intel QAT engine container
+
+There is an OpenSSL Intel QAT Dockerfile located at 
+[https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine](https://github.com/intel/intel-device-plugins-for-kubernetes/tree/master/demo/openssl-qat-engine). 
```

Comment:
Another "double URL link" to replace with either the URL or change the name to some text to describe the link.

Reply:
Done